### ISSUE TITLE
[Deprecate]: Determine if netlify-cms-backend-fs should be deprecated

### ISSUE THREAD

**Issue by talves** [ADARTA/netlify-cms-components#21]
It looks like `netlify-cms` has started working on getting a backend solution for file-system up and running: https://github.com/netlify/netlify-cms/tree/master/packages/netlify-cms-proxy-server

This would remove the need for this library to continue to be maintained, if it is a viable solution.

It looks like it will default to file-system, then have the ability to use/change the local git repository. This is speculation at this point, because it is not very clear in the docs yet.

**Comment by smashercosmo** [comment]
Yeah, let's see what they will come up with. Your solution works amazingly well. And smoothly integrates with gatsby. I didn't have any problems with it for a year (until they made these internal breaking changes). From the first glance and don't like that you need to start netlify-cms-proxy-server in the separate process. I would prefer to have it as a middleware for gatsby server.

**Comment by smashercosmo** [comment]
Played a bit with netlify-cms-proxy-server. It works fine. Didn't notice any issues. Though as I said before, it's a bit annoying to start separate process. But they're saying that they are going to create different SSG plugins for that purpose. https://github.com/netlify/netlify-cms/issues/2335#issuecomment-585808063

**Comment by talves** [comment]
@smashercosmo Thanks for the update. I'm working on Gatsby a lot lately without setting up netlify-cms yet, so will probably circle around to see how integrated it is also. I do agree, it would have been nice if the server was in a middleware layout like I have here. Over complicating it is kind of par for the course for netlify-cms due to the setup. If I don't like it, I will just fix this library.

**Comment by d4rekanguok** [comment]
Erez shared this bit from Slack, I'd figure to post it here:


> This should work:
> ```js
> const { registerLocalFs } = require('netlify-cms-proxy-server/dist/middlewares');
> exports.onCreateDevServer = async ({ app }) => {
>   await registerLocalFs(app);
> };
> ```
>
> Make sure to update your config.yml accordingly:
> ```yml
> local_backend:
>   url: http://localhost:8000/api/v1
> ```

**Comment by smashercosmo** [comment]
Doesn't work for me. Can't get past identity widget. When using separate proxy server with ```npm run netlify-cms-proxy-server``` admin dashboard opens without any request for credentials.

**Comment by smashercosmo** [comment]
Ok, figured it out. Instead of ```http://localhost:8000/api``` use ```http://localhost:8000/api/v1```

**Comment by talves** [comment]
ok, still have not had time. How do you feel about the solution so far?

Let me know if it seems like a viable replacement to what I solved here. If so, I will move it to my lower priority list. At first glance it just seemed over complicated for something that is dev driven, and couldn't understand why it was chosen to add that extra code for a solution that was local dev only.

To be fair, I haven't reviewed the code yet, so I'm not sure how it ended up, I just hope it's not included into the build bundle and allows/requires it to be configured during development only.

**Comment by d4rekanguok** [comment]
@talves Your backend-fs was a game changer, I wish it was incorporated into an official solution. I have only used `netlify-cms-proxy-server` briefly, but see no changes in experience so far.

@smashercosmo thanks for figuring that out, I haven't tried it yet but now I know what to do :+1:

**Comment by talves** [comment]
@smashercosmo , @d4rekanguok thanks for your patience on this. Finally had time this weekend to do some review. Also thanks for the kind words and support on this library.

(TL;DR) The new "proxy" server is viable as a solutions and don't see a reason to not use it. I will be deprecating this library and pointing to the official solution. I will not be integrating `netlify-cms` into my Gatsby site or any React app going forward though as I point out below why. 

I'm surprised that after all the prodding to include this library as an official `netlify-cms` package, the final solution ended  up being the same processes this library adhered to for the FS piece as a middleware to an express server. Anyway, I'm glad to see the final solution allows for the ignoring of the local git solution now and FINALLY splits out the two. I feel it should not be included in the build bundle of `netlify-cms-app` but to be fair, I believe that should be the case for all the backends. There's no reason to include them by default when they once could easily be added as registered backends. Currently the `umd` builds are not being included in the libraries and is really interesting to me, because it seemed to be such an important thing when I worked on the monorepo to fix the whole build process for a single react instance. (**note:** there was discussion for that to be the case in 3.0, and I recommend that be done still).

Gatsby has a solution now as your comments above were so kind to be included here for reference. **I've decided to NOT include the cms using `gatsby-plugin-netlify-cms` to my sites. I thought this had been fixed earlier, but maybe it was never fixed. ¯\\_(ツ)_/¯ **I use style libraries that use `emotion` and `netlify-cms` is breaking those sites by overlaying my `GlobalStyles` component state which I warned about a couple years ago when discussing styling. State context for `netlify-cms` doesn't adhere to being it's own component library and don't see it doing that without a full rewrite. It might work if it was hoisted up to the `netlify-cms` bundle. I can't spend the time trying to fix all the issues, so I will create a separate React app for `netlify-cms` and probably host it on a sub-domain or some other domain.

This brings me back to the decisions for the "proxy server". For local development, a new process must be created with an express server to connect to the `api`. This design choice was a requirement for this library and also ended up being the same solution for `netlify-cms` local development using the "proxy server", so in that case both solutions are the same and my reason for **deciding to deprecate this library**. 😁 Better to be maintained as an official solution where the api will get fixed when it changes as what has happened here. Keeping the API in beta and allowing it to break without a breaking change version bump is just bad unfortunately.
